RoboticonColony/Assets/Code/Timeout.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Diagnostics;
+
+public class Timeout
+{
+    private int timeAllowed;
+    private Stopwatch stopwatch = new Stopwatch();
+
+    /// <summary>
+    /// creates a timout object. use to keep track of how long phases are going
+    /// </summary>
+    /// <param name="time">time in seconds of the timeout</param>
+    public Timeout(int time)
+    {
+        if (time < 1)
+        {
+            throw new ArgumentOutOfRangeException("Must be at least one second");
+        }
+        timeAllowed = time;
+    }
+
+    /// <summary>
+    /// start the timer
+    /// </summary>
+    public void Start()
+    {
+        stopwatch.Start();
+    }
+
+    /// <summary>
+    /// get the seconds remaining before a timeout
+    /// </summary>
+    public int SecondsRemaining
+    {
+        get
+        {
+            return Math.Max(timeAllowed - stopwatch.Elapsed.Seconds, 0);
+        }
+    }
+
+    /// <summary>
+    /// check if the timeout has reached its time allowed
+    /// </summary>
+    public bool Finished
+    {
+        get
+        {
+            return timeAllowed - stopwatch.Elapsed.Seconds > 0;
+        }
+    }
+}
```

Comment:
I think you need to add a stopwatch.Stop() method somewhere in finished, else when you start it again for the next time a player does that phase, it will still be at the value it was. I could be mistaken though, never personally used Stopwatch.